extensions/assets-manager/AssetsManagerEx.cpp

```diff
@@ -476,11 +498,14 @@ void AssetsManagerEx::parseVersion()
         }
         else
         {
+            // Save the old update entry type since its value may be changed in event listener.
+            UpdateEntry oldUpdateEntry = _updateEntry;
+
             _updateState = State::NEED_UPDATE;
             dispatchUpdateEvent(EventAssetsManagerEx::EventCode::NEW_VERSION_FOUND);
 
             // Wait to update so continue the process
-            if (_waitToUpdate)
+            if (oldUpdateEntry == UpdateEntry::DO_UPDATE)
```

Comment:
If the old entry was CHECK_UPDATE then it will be set to NONE, why not check 

if (_updateEntry == UpdateEntry::DO_UPDATE) here ?